data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/Es6IndexTemplateAPIWrapper.java

```diff
@@ -0,0 +1,134 @@
+import org.opensearch.client.opensearch.indices.TemplateMapping;
+import org.opensearch.client.transport.JsonEndpoint;
+import org.opensearch.client.transport.endpoints.BooleanResponse;
+import org.opensearch.client.transport.endpoints.SimpleEndpoint;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class Es6IndexTemplateAPIWrapper implements IndexTemplateAPIWrapper<TemplateMapping> {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final OpenSearchClient openSearchClient;
+
+    public Es6IndexTemplateAPIWrapper(final OpenSearchClient openSearchClient) {
+        this.openSearchClient = openSearchClient;
+    }
+
+    @Override
+    public void putTemplate(final IndexTemplate indexTemplate) throws IOException {
+        if(!(indexTemplate instanceof LegacyIndexTemplate)) {
+            throw new IllegalArgumentException("Unexpected indexTemplate provided to createTemplate.");
+        }
+
+        final Map<String, Object> templateMapping = ((LegacyIndexTemplate) indexTemplate).getTemplateMap();
+        final String indexTemplateString = OBJECT_MAPPER.writeValueAsString(templateMapping);
+
+        // Parse byte array to Map
+        final ByteArrayInputStream byteIn = new ByteArrayInputStream(
+                indexTemplateString.getBytes(StandardCharsets.UTF_8));
+        final JsonpMapper mapper = openSearchClient._transport().jsonpMapper();
+        final JsonParser parser = mapper.jsonProvider().createParser(byteIn);
+
+        final PutTemplateRequest putTemplateRequest = PutTemplateRequestDeserializer.getJsonpDeserializer()
+                .deserialize(parser, mapper);
+
+        final OpenSearchIndicesClient openSearchIndicesClient = openSearchClient.indices();
+        final JsonEndpoint<PutTemplateRequest, PutTemplateResponse, ErrorResponse> endpoint = es6PutTemplateEndpoint(putTemplateRequest);
+        openSearchIndicesClient._transport().performRequest(putTemplateRequest, endpoint, openSearchIndicesClient._transportOptions());
+    }
+
+    @Override
+    public Optional<TemplateMapping> getTemplate(final String templateName) throws IOException {
+        final ExistsTemplateRequest existsTemplateRequest = new ExistsTemplateRequest.Builder()
+                .name(templateName)
+                .build();
+        final BooleanResponse booleanResponse = openSearchClient.indices().existsTemplate(
+                existsTemplateRequest);
+        if (!booleanResponse.value()) {
+            return Optional.empty();
+        }
+
+        final GetTemplateRequest getTemplateRequest = new GetTemplateRequest.Builder()
+                .name(templateName)
+                .build();
+        final GetTemplateResponse response = openSearchClient.indices().getTemplate(getTemplateRequest);
+
+        if (response.result().size() == 1) {
```

Comment:
Should this logic be moved into `V1TemplateStrategy`? It seems this is related to the model, not the wire model. So it would be appropriate there instead of here.